spark-tpcds-benchmark-runner/src/main/java/com/palantir/spark/tpcds/datagen/GenSortDataGenerator.java

```diff
@@ -65,58 +66,34 @@ public GenSortDataGenerator(
-        if (tempWorkingDir.toFile().isDirectory()) {
-            FileUtils.deleteDirectory(tempWorkingDir.toFile());
-        }
-        if (!tempWorkingDir.toFile().mkdirs()) {
-            throw new IllegalStateException(
-                    String.format("Could not create temporary work directory at %s", tempWorkingDir));
-        }
         try {
-            final Path genSortFilePath = extractBinary();
-            log.info("Extracted gensort binary: {}", SafeArg.of("genSortFilePath", genSortFilePath));
-            Path dataDir = Files.createDirectory(tempWorkingDir.resolve(GENSORT_DATA_DIR_NAME));
-            Process gensortProcess = new ProcessBuilder()
-                    .command(
-                            genSortFilePath.toFile().getAbsolutePath(),
-                            "-a",
-                            Long.toString(numRecords),
-                            dataDir.resolve(GENERATED_DATA_FILE_NAME + ".csv")
-                                    .toAbsolutePath()
-                                    .toString())
-                    .inheritIO()
-                    .directory(genSortFilePath.toFile().getParentFile())
-                    .start();
-            int returnCode = gensortProcess.waitFor();
-            if (returnCode != 0) {
-                throw new IllegalStateException(String.format("genSort failed with return code %d", returnCode));
+            if (tempWorkingDir.toFile().isDirectory()) {
+                FileUtils.deleteDirectory(tempWorkingDir.toFile());
             }
-            log.info("Finished running gensort");
-            DataGenUtils.uploadFiles(
-                    destinationFileSystem,
-                    paths.csvDir(SCALE),
-                    dataDir.toFile(),
-                    MoreExecutors.newDirectExecutorService());
-            StructType schema = DataTypes.createStructType(
-                    ImmutableList.of(DataTypes.createStructField("record", DataTypes.StringType, false)));
-            String destinationPath =
-                    Paths.get(paths.parquetDir(SCALE), GENERATED_DATA_FILE_NAME).toString();
-            parquetTransformer.transform(
-                    spark, schema, paths.tableCsvFile(SCALE, GENERATED_DATA_FILE_NAME), destinationPath, "\n");
-            registration.registerTable("gensort_data", schema, SCALE);
+            if (!tempWorkingDir.toFile().mkdirs()) {
+                throw new IllegalStateException(
+                        String.format("Could not create temporary work directory at %s", tempWorkingDir));
+            }
+
+            Path genSortBinaryPath = extractBinary();
+            log.info("Extracted gensort binary: {}", SafeArg.of("genSortBinaryPath", genSortBinaryPath));
+            Path dataDir = Files.createDirectory(tempWorkingDir.resolve(GENSORT_DATA_DIR_NAME));
+            scales.forEach(scale -> generateData(dataDir, scale, genSortBinaryPath));
```

Comment:
we can parallelize this similar to the TpcdsDataGenerator, will do in a follow up